packages/components/src/pages/ChatSearch.vue

```diff
@@ -254,6 +250,24 @@ export default {
     },
   },
   methods: {
+    // TODO: This is a suggestion of how thread loading could work.
+    // TODO: Listen for the threadId from the chat component and emit a corresponding message.
+    async loadThread(threadId: string) {
+      const thread = await this.rpcClient.loadThread(threadId);
+      const chat = this.$refs.vchat;
+
+      for (const message of thread.messages) {
+        if (message.role === 'user') {
+          const userMessage = chat.addUserMessage(message);
+          if (message.codeSelection) userMessage.codeSelections = [message.codeSelection];
+          // TODO: User message provides prompt, but the UI does not have a place for this.
+        } else {
+          const systemMessage = chat.addSystemMessage(message);
```

Comment:
Double check the arguments here for `addUserMessage`/`addSystemMessage`. I don't think they accept a full message structure, though I wouldn't be opposed to that. This is the downside of `//@ts-nocheck`.